x-pack/legacy/plugins/maps/public/embeddable/map_embeddable.js

```diff
@@ -244,5 +257,17 @@ export class MapEmbeddable extends Embeddable {
         openTOCDetails,
       });
     }
+
+    const hiddenLayerIds = this._store
```

Comment:
Can you please move this logic to a selector in https://github.com/elastic/kibana/blob/master/x-pack/legacy/plugins/maps/public/selectors/map_selectors.js